src/relations/db.py

```diff
@@ -133,11 +133,19 @@ def __init__(self, charm: CharmBase, admin: bool = False):
         self.charm = charm
         self.admin = admin
 
+    def _depart_flag(self, relation):
+        return f"{self.relation_name}_{relation.id}_departing"
+
     def _on_relation_joined(self, join_event: RelationJoinedEvent):
         """Handle db-relation-joined event.
 
         If the backend relation is fully initialised and available, we generate the proposed
         database and create a user on the postgres charm, and add preliminary data to the databag.
+
+        Deferrals:
+            - If backend is unavailable
+            - If pgbouncer has not started
+            - If database and password haven't been added to the databag by the client charm
```

Comment:
This comment makes me think that the password is added by the client charm, but in fact it is added by the leader unit in the peer relation.